react/game/src/redux/auth/actions.js

```diff
@@ -1,54 +1,116 @@
-import { service as authService, loadAuthState, saveAuthState } from '@services/authService';
+import { service as authService, loadAuthState, saveAuthState, deleteAuthState } from '@services/authService';
+import { deleteGameState } from '@services/gameService';
+import { API } from '@config/api';
+import { push } from 'connected-react-router';
 
-export const LOGIN_ACTIONS = {
+export const AUTH_ACTIONS = {
   LOAD_APP: 'LOAD_APP',
   APP_LOADED: 'APP_LOADED',
   LOGIN_LOADING: 'LOGIN_LOADING',
   LOGIN_ERROR: 'LOGIN_ERROR',
-  LOGIN_SUCCESS: 'LOGIN_SUCCESS'
+  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
+  SET_USER_INFO: 'SET_USER_INFO',
+  LOGOUT: 'LOGOUT'
 };
 
 const privateActionCreators = {
-  loadApp: session => ({
-    type: LOGIN_ACTIONS.LOAD_APP,
-    payload: session
+  loadApp: ({ userId, token }) => ({
+    type: AUTH_ACTIONS.LOAD_APP,
+    payload: { userId, token }
   }),
-  appLoaded: loaded => ({
-    type: LOGIN_ACTIONS.APP_LOADED,
-    payload: { loaded }
+  appLoaded: appLoaded => ({
+    type: AUTH_ACTIONS.APP_LOADED,
+    payload: { appLoaded }
   }),
   loginLoading: isLoading => ({
-    type: LOGIN_ACTIONS.LOGIN_LOADING,
+    type: AUTH_ACTIONS.LOGIN_LOADING,
     payload: { isLoading }
   }),
   loginError: errorMessage => ({
-    type: LOGIN_ACTIONS.LOGIN_ERROR,
+    type: AUTH_ACTIONS.LOGIN_ERROR,
     payload: { errorMessage }
   }),
-  loginSuccess: (id, token) => ({
-    type: LOGIN_ACTIONS.LOGIN_SUCCESS,
-    payload: { id, token }
+  loginSuccess: (userId, token) => ({
+    type: AUTH_ACTIONS.LOGIN_SUCCESS,
+    payload: { userId, token }
+  }),
+  setUserInfo: userInfo => ({
+    type: AUTH_ACTIONS.SET_USER_INFO,
+    payload: { userInfo }
+  }),
+  logout: () => ({
+    type: AUTH_ACTIONS.LOGOUT
   })
 };
 
 export const actionCreators = {
-  initApp: () => dispatch => {
+  initApp: () => async dispatch => {
     dispatch(privateActionCreators.appLoaded(false));
+    dispatch(privateActionCreators.loginLoading(true));
     const session = loadAuthState();
-    const { token, id } = session || { token: null, id: null };
-    dispatch(privateActionCreators.loadApp({ token, id }));
-    dispatch(privateActionCreators.appLoaded(true));
+    const { token, userId } = session || { userId: null, token: null };
+    dispatch(privateActionCreators.loadApp({ userId, token }));
+    if (session) {
+      API.setHeader('Authorization', session.token);
+      const responseUser = await authService.get(session.userId);
+      dispatch(privateActionCreators.loginLoading(false));
+      const userData = responseUser.data;
+      if (responseUser.ok) {
+        const userInfo = {
+          id: userData.id,
+          email: userData.email,
+          firstName: userData.firstName,
+          lastName: userData.lastName,
+          age: userData.age
+        };
+        dispatch(privateActionCreators.setUserInfo(userInfo));
```

Comment:
Why not just ...
```suggestion
        dispatch(privateActionCreators.setUserInfo({ ...userData });
```
So u avoid to create `userInfo` const

Reply:
👌🏼